### DISCUSSION TITLE
How to evaluate whenLoaded only when the relationship is preset?

### DISCUSSION THREAD

**Discussion by benjivm** [spatie/laravel-data#100] [Q&A] [answered]
I have the following property in a data class:

```php
public Lazy|ModelVersionData|null $modelVersion,
```

It's a `belongsTo` relationship that is sometimes loaded, and may not exist even when loaded.

When I use the following Lazy loader I received an error:

```php
Lazy::whenLoaded('modelVersion', $model, fn () => ModelVersionData::from($model->modelVersion)),

// Error:
// Could not create a `App\Data\ModelVersionData` object from value with type `NULL`
```

So I have to write it as this:

```php
Lazy::whenLoaded('modelVersion', $model, fn () => $model->modelVersion ? ModelVersionData::from($model->modelVersion) : null),
```

Is there a way to make this work more like [Laravel's Resources](https://laravel.com/docs/9.x/eloquent-resources#conditional-relationships), where it simply returns `null` if the relationship is empty, even when `whenLoaded` is `true`, or am I perhaps doing something wrong?

**Comment by rubenvanassche** [comment, marked as answer]
Hi @benjivm,

This should be fixed in the release later today!
